ft232/mpsse.py

```diff
@@ -0,0 +1,555 @@
+        Returns
+        -------
+        bool
+        """
+        return self.__Ft232.MPSSE_CLOCK == 60e6
+
+    def __highByteExists(self):
+        """ This checks if the board has more than 8 pins. 
+        
+        Returns
+        -------
+        bool
+        """
+        return (len(self.__Ft232.PortC) >= 1)
+
+    def __isPinLowByte(self, Pin):
+        """ This checks if the Pin is one of the first 8 pins (LowBytePin).
+        
+        Returns
+        -------
+        bool
+        """
+        return (0 <= Pin <= 7)
+
+    def __isPinHighByte(self, Pin):
+        """ This checks if the Pin is one of the pins 9 ++ (HighBytePin).
+        
+        Returns
+        -------
+        bool
+        """
+        return self.__highByteExists() and (8 <= Pin <= (len(self.__Ft232.PortC) + 7))
+
+    def __setPinConfig(self):
+        """ Sets the __PinValues and __PinDirections to the device. 
+        """
+        Data = []
+        Data.append(self.__pinConfigCommand(MPSSE.CONFIG_LOW_PINS, MPSSE.SET_PIN_CONFIG))
+        Data.append(self.__PinValues[0])
+        Data.append(self.__PinDirections[0])
+        ## Checks if the board has more than 8 Pins
+        if self.__highByteExists():
+            Data.append(self.__pinConfigCommand(MPSSE.CONFIG_HIGH_PINS, MPSSE.SET_PIN_CONFIG))        
+            Data.append(self.__PinValues[1])
+            Data.append(self.__PinDirections[1])
+        self.__Ft232.write(bytes(Data))
+
+    def __getPinConfig(self):
+        """ Gets the actual PinValues from the device as a list.
+        
+        Returns
+        -------
+        list
+            The PinValues as a list of bytes
+        """
+        Command = []
+        Command.append(self.__pinConfigCommand(MPSSE.CONFIG_LOW_PINS, MPSSE.GET_PIN_CONFIG))
+        ## Checks if the board has more than 8 Pins
+        if self.__highByteExists():
+            Command.append(self.__pinConfigCommand(MPSSE.CONFIG_HIGH_PINS, MPSSE.GET_PIN_CONFIG))
+        self.__Ft232.write(bytes(Command))
+        return self.__Ft232.read(len(Command))
+
+    def __setPinDirection(self, Pin, InputOutputPin):
+        """ Make a Pin an input or output. This does not actually set the pin physically. Use _setPinDirection() or _setPinsDirection() instead. Otherwise call _setPinConfig() after using this function to set the direction physically.
+        Parameters
+        ----------
+        Pin : int, 4 - 15
+            The Pin whose direction should be set.
+        
+        InputOutputPin : GPIO.INPUT / GPIO.OUTPUT
+            The direction the Pin is set to.
+        """
+        if (Pin >= self.__Ft232.D3) and self.__isPinLowByte(Pin):
+            self.__PinDirections[0] = (self.__PinDirections[0] & ~(1 << Pin)) | (InputOutputPin << Pin)
+        elif self.__isPinHighByte(Pin):
+            self.__PinDirections[1] = (self.__PinDirections[1] & ~(1 << (Pin - 8))) | (InputOutputPin << (Pin - 8))
+        else:
+            print("Pin is out of range")
```

Comment:
The library should never print. You should probably raise an exception for an error.

Reply:
Yes, i think i missed a few of those prints which should be exceptions.